### PR TITLE
Fixing rounding errors when inserting microseconds

### PR DESCRIPTION
**Pull request by AndreCAndersen**
When calculating nanoseconds since EPOCH, `total_seconds()` will give a float representation which when multiplied by 1e9 and cast to `int` will create rounding errors. E.g. inserting `2017-01-17T08:00:12.000001` with a datetime will produce `2017-01-17T08:00:12.000001024Z` in InfluxDB. This commit fixes this by never going via floating seconds, and is based on the `total_seconds()` function itself.

### CODE REVIEW

File: influxdb/line_protocol.py [hunk 1]
```diff
@@ -16,6 +16,12 @@
 EPOCH = UTC.localize(datetime.utcfromtimestamp(0))
 
 
+def _to_nanos(timestamp):
+    delta = timestamp - EPOCH
+    nanos = (delta.days * 86400 + delta.seconds) * 10 ** 9 + delta.microseconds * 10 ** 3
```
Comment (xginn8) [inline influxdb/line_protocol.py]: this line fails flake8 (89 chars), mind splitting it up? otherwise, this looks like a good change.

Comment (AndreCAndersen) [inline influxdb/line_protocol.py]: [How about now](https://github.com/influxdata/influxdb-python/pull/407/commits/01bc900a9e2f4dde1471f56cdcf34a9394e04d1e)?

**Comment by clslgrnc** [comment]
Just encountered this rounding error and was about to submit this very patch.
Can this be merged?

In order to ensure that ns stays an int I would also rewrite:
```python
        if precision is None or precision == 'n':
            return ns
        elif precision == 'u':
            return ns / 1e3
        elif precision == 'ms':
            return ns / 1e6
        elif precision == 's':
            return ns / 1e9
        elif precision == 'm':
            return ns / 1e9 / 60
        elif precision == 'h':
            return ns / 1e9 / 3600
```

to
```python
        if precision is None or precision == 'n':
            return ns
        elif precision == 'u':
            return ns / 10**3
        elif precision == 'ms':
            return ns / 10**6
        elif precision == 's':
            return ns / 10**9
        elif precision == 'm':
            return ns / 10**9 / 60
        elif precision == 'h':
            return ns / 10**9 / 3600
```

**Comment by clslgrnc** [comment]
I would probably allow you to close some of those:
https://github.com/influxdata/influxdb-python/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+nanoseconds

**Comment by sebito91** [comment]
Duplicate of #649 and fixed in #650

**Comment by sebito91** [comment]
Re-addressed in a new PR that merges this content including fixes from @clslgrnc

**Pull request closed without merging**